modules/oe_theme_helper/src/TwigExtension/TwigExtension.php

```diff
@@ -490,4 +494,37 @@ public function toEclIcon(array $context, $icon): array {
     ];
   }
 
+  /**
+   * Trims the contents of a text field.
+   *
+   * The method expects to receive a string (already rendered).
+   * Inside a template we can use '|render' for passing to filter
+   * only string like in example: {{ long_text|render|smart_trim(10) }}.
+   *
+   * @param mixed $text
+   *   The text to be trimmed.
+   * @param int $limit
+   *   Amount of text to allow.
+   *
+   * @return string|array
+   *   The trimmed text.
+   */
+  public function smartTrim($text, $limit = NULL) {
+    // Return original text if length is not available.
+    if ($limit === NULL) {
+      return $text;
+    }
+
+    $truncate = new TruncateHTML();
+
+    if (is_string($text) || $text instanceof MarkupInterface) {
+      // If $text is a string or a Markup object, trim
+      // and return the Markup object.
+      return Markup::create($truncate->truncateChars($text, $limit));
```

Comment:
If we get a string we shouldn't turn that into a markup, as this would mean that a string with `<javascript>...</javascript>` would suddenly become executable code.

Let's simply do
```php
public function smartTrim($text, $limit) {
  $truncate = new TruncateHTML();
  return $truncate->truncateChars($text, $limit);
}
```
truncateChars handles all cases anyway, then let's have a test that ensure that passing a MarkupInterface behaves correctly.

Reply:
That didn't work because then the string would be stripped when rendered. We then changed approach, the only way is to respect the input.